fix: Remove horizontal scroll from match vouchers tab

marination opened the pull request:
Solves https://github.com/alyf-de/banking/issues/48

- Adjusted horizontal space between transactions list and actions area
- Reduced scrollbar idth of transactions area
- Adjusted column widths of datatable
- Document Type field moved to end of datatable. Its not an important column and often has repetitive information

## Before:
- The important columns are not all visible
- LHS is has some whitespace since its taking up way more space than it needs
- Horizontal scroll
<img width="1315" alt="Screenshot 2023-11-03 at 12 24 28 PM" src="https://github.com/alyf-de/banking/assets/25857446/82ca129d-ae65-40bf-93ac-5d75d3f92592">


## After:
- Match reasons are highlighted in bold. One less action for users
- No horizontal scroll
- Reference, Party and Voucher are left aligned
<img width="1306" alt="Screenshot 2023-11-06 at 4 03 59 PM" src="https://github.com/alyf-de/banking/assets/25857446/4e3caf3e-52e9-462b-bd8d-218868ac54e1">

marination committed:
fix: Remove horizontal scroll from match vouchers tab

- Adjusted horizontal space between transactions list and actions area
- Reduced scrollbar idth of transactions area
- Adjusted column widths of datatable
- Document Type field moved to end of datatable. Its not an important column and often has repetitive information

barredterra commented on banking/public/js/bank_reconciliation_beta/actions_panel/match_tab.js:
```diff
@@ -473,11 +468,16 @@ erpnext.accounts.bank_reconciliation.MatchTab = class MatchTab {
+					return frappe.format(value, {fieldtype: "Link", options: data[6]});
 				},
 			},
+			{
+				name: __("Document Type"),
+				editable: false,
+				width: 50,
+			},
```
Could we even set this to hidden and only use it for formatting the link in Document Name? Usually the doctype should be obvious from the naming series.

barredterra committed:
fix: store amount as number, not string

barredterra committed:
feat: formatted docname instead of doctype column

barredterra committed:
feat: link to party

barredterra committed:
refactor: fluid layout and shorter column labels

barredterra committed:
refactor: remove index column

barredterra commented:
I pushed the cleanup a bit further:

- Format remaining amount only for display, not storage (#79)
- Format docname as link, remove doctype column
- Format party as link
- Now we have more space and I didn't want to fine tune the widths again, so I changed the layout to "fluid"
- Removed the index column for even more space

![Bildschirmfoto 2023-11-05 um 20 14 55](https://github.com/alyf-de/banking/assets/14891507/ad7da43a-75a8-4a0d-abd8-05e3151a347b)

Remaining issues (can be solved in a separate PR):

- [x] Reference, Party and Voucher should be left aligned
- [x] Format individual cells as bold or with background to show which values matched, remove "Reason" column

Feel free to merge

marination committed:
Merge branch 'version-14' into remove-horizontal-scroll

marination committed:
fix: Remove Match Reasons and cell value alignment

- Avoid passing values to column formatter. Format at cell level. Auto-fixes alignment for non text columns
- Specify column alignment for text column Refernece No
- Remove Match Reasons button. Highlight matched values in bold

marination commented:
Updated description

marination committed:
fix: Adjust indices as a column was removed
